apps/angular/permissions/src/app/login.component.ts

```diff
@@ -35,29 +39,20 @@ import { UserStore } from './user.store';
       Enter application
     </button>
   `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class LoginComponent {
-  constructor(private userStore: UserStore) {}
+  private readonly userStore = inject(UserStore);
 
-  admin() {
-    this.userStore.add(admin);
-  }
-  manager() {
-    this.userStore.add(manager);
-  }
-  reader() {
-    this.userStore.add(reader);
-  }
-  writer() {
-    this.userStore.add(writer);
-  }
-  readerWriter() {
-    this.userStore.add(readerAndWriter);
-  }
-  client() {
-    this.userStore.add(client);
-  }
-  everyone() {
-    this.userStore.add(everyone);
+  readonly admin = admin;
+  readonly manager = manager;
+  readonly reader = reader;
+  readonly writer = writer;
+  readonly readerAndWriter = readerAndWriter;
+  readonly client = client;
+  readonly everyone = everyone;
+
+  select(role: User): void {
+    this.userStore.select(role);
```

Comment:
a nicer way to do something similar would be to be:
select accept a union type, than you create a map between your union type and the wanted object. 👍

Reply:
updated in [here](https://github.com/tomalaforge/angular-challenges/pull/623/commits/8c40ac42f938be2c057283e2bffec844e01bda88)